.github/workflows/test-go-latest.yaml

```diff
@@ -63,19 +71,45 @@ jobs:
         key: go-govendor-{{ hashFiles('**/vendor.json') }}
     - name: Get Go dependencies
       run: cd ${{ github.workspace }}/src/github.com/snapcore/snapd && ${{ github.workspace }}/bin/govendor sync
+    - name: Prepare results store
+      run: mkdir "${{ github.workspace }}/.test-results"
     - name: Run static checks
-      run: cd ${{ github.workspace }}/src/github.com/snapcore/snapd && ./run-checks --static
+      run: |
+        stamp="${{ github.workspace }}/${{ github.run_id }}-${{ github.job }}-${{ github.action }}-success"
```

Comment:
Given that this is outside of the the directory you're caching, when would you ever expect this to exist prior to the `touch` command below?  And is there any reason to include `github.run_id` and `github.job` in the stamp file name if those are part of the cache key?

Reply:
The intent was to leave a stamp when the tests pass and have it picked up by the cache action. The next time you have to restart the whole workflow because of some random spread failure, the stamps get restored from the cache and the step does not need to be repeated again. The key includes run_id to make the stamps cache specific for one run, which if I understand correctly, does not change across restarts. The job_id is included because I wanted to use that in spread job.

Anyways, this needs changes because the cache is only posted when a job is successful.

Reply:
My point is that you are caching `$GITHUB_WORKSPACE/.test-results`, but your stamp file is not under `$GITHUB_WORKSPACE/.test-results`.  So your stamp file will never be saved or restored by the cache action.

As far as naming the stamp files, if each (run_id, job_id) has its own cache, you don't need to include that information in the stamp file name.  One run or job won't see another run or job's stamp files, so it doesn't matter if they use the same file name.

Reply:
Hah, thanks for catching this!